### PR TITLE
Allow csv query param

### PR DESCRIPTION
**Pull request by nikhiljindal**
Forked from https://github.com/kubernetes/kubernetes/pull/12507

Updates the server to accept multiple values of a query param in csv form (exec?command=echo,Hi) in addition to the current way of requiring multiple instances of the param (exec?command=echo&command=Hi)
This is because swagger 1.0 has no way of defining a query param that accepts multiple instances. It only supports multiple values in the form of csv using the allowMultiple property(https://github.com/swagger-api/swagger-spec/blob/master/versions/1.2.md#524-parameter-object)
Swagger 2.0, however, supports that with collectionFormat using multi:

```
multi - corresponds to multiple parameter instances instead of multiple values 
for a single instance foo=bar&foo=baz. This is valid only for parameters in 
"query" or "formData"
```

So we can just wait for go-restful to support 2.0 or accept this PR.

Without a fix, clients generated using swagger spec will not be able to pass multiple command arguments.

cc @bgrant0607 @lavalamp @smarterclayton

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Ignore the first commit. That will go away once https://github.com/kubernetes/kubernetes/pull/12507 is merged

**Comment by smarterclayton** [comment]
Will `exec?command=echo,hi` properly handle a command like `["some
command", "\"another,arg\""]`?

On Fri, Aug 21, 2015 at 4:28 PM, Nikhil Jindal <notifications@github.com>
wrote:

> Ignore the first commit. That will go away once #12507
> <https://github.com/kubernetes/kubernetes/pull/12507> is merged
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13054#issuecomment-133551170>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift

File: pkg/runtime/conversion_test.go [hunk 1]
```diff
@@ -55,6 +57,16 @@ func TestStringMapConversion(t *testing.T) {
 		errFn    func(error) bool
 		expected runtime.Object
 	}{
+		"array works": {
+			input: map[string][]string{
+				"String":    {"not_used"},
```
Comment (smarterclayton) [inline pkg/runtime/conversion_test.go]: Need to handle quotes, escape characters, and arguments with commas in them (and all three).  This won't be safe to allow unless you can enforce that someone can't do an argument injection attack on the `exec` command query.

I.e. consider all query values tainted, ensure that there can never be a command argument injection attack in this code path.  Which means client and server have to define escaping rules for commas and quotes.

**Comment by smarterclayton** [comment]
I'm not sure we can make this sufficiently safe that a client won't accidentally screw it up.  I don't think swagger is important enough to risk command injection attacks from naive clients.

**Comment by bgrant0607** [comment]
I have similar concerns about quoting, etc.

We may just need to generate the swagger as if we supported CSV and then just document that we take multiple args instead.

And we should probably file an issue for swagger 2.0 support and put help-wanted on it. And we'll need to decide on a versioning scheme for the swagger API endpoint.

**Comment by lavalamp** [comment]
I think there's actually no reasonable general way to know what characters indicate quotation and what characters don't: it's highly data specific. So I think that any generalized CSV format is Doomed, and multiple args is a much more natural way to do this.

**Pull request closed without merging**